odoo/osv/expression.py

```diff
@@ -571,7 +573,7 @@ def parent_of_domain(left, ids, left_model, parent=None, prefix=''):
                 records = left_model.sudo().browse(ids)
                 while records:
                     parent_ids.update(records._ids)
-                    records = records[parent_name]
+                    records = records[parent_name] - records.browse(parent_ids)
```

Comment:
Hello @rco-odoo 

we need to fix potencial infinite loop, which may happen when `child_of` is used by mistake. 

It was reported by customer and I also found  the same issue in `hr_timesheet` module (fixed as part of this PR)